doc/manual/src/design/store/paths.md

```diff
@@ -0,0 +1,51 @@
+# Store Paths
+
+A store path is a pair of a 20-byte digest and a name.
+
+Historically it is the triple of those two and also the store directory, but the modern implementation's internal representation is just the pair.
+This change is because in the vast majority of cases, the store dir is fully determined by the context in which the store path occurs.
+
+## String representation
+
+A store path is rendered as the concatenation of
+
+  - the store directory
```

Comment:
I will expect what this is before. Truth be told, there is no one universal answer, because not all stores even mount stuff, and thus we don't always need this.

The "truth" revealed at the end of the chapter in the "relocation" section is store objects conceptually do have a store path if and only if they have self references.